building/configuring_git_workflow.adoc

```diff
@@ -0,0 +1,426 @@
+
+# commit changes
+git commit -m "JBIDE-12345 fix that thing that was broken" somefile
+git commit -m "JBIDE-12345 additional fix" somefile
+git commit -m "typo" somefile
+
+# squash commits?
+git rebase -i HEAD~3 # eg., if you want to squash 3 commits into a single one
+
+# rebase your changes against the origin
+git pull --rebase origin ${branch}
+
+# push to your fork
+# ${gituser} is your username at github
+git push ${gituser} ${topic}
+
+# create pull request
+# if you use hub (see `Submitting Patches` above), you can create a pull request via commandline:
+parentProject=`git remote -v | grep origin | grep push | sed "s/.\+github.com\(:\|\/\)\(.\+\)\/.\+/\2/"`
+hub pull-request -o -f -m "`git log -1 --pretty=%B` >> ${branch}" -b ${parentProject}:${branch} -h ${gituser}:${topic}
+----
+
+== Accepting Patches (Maintainers)
+
+Maintainers and committers are required to do due dilligence on any patches that come their way. 
+This includes an initial *code review* to catch any obvious problems, as well as *smoke tests*.
+Maintainers may require patches come with *unit tests* as well, so they can easily verify that the
+given patch does in fact fix the given issue. 
+
+To check out a given contribution and run the tests or manually smoke-test the contribution, 
+maintainers should look at the pull request to determine 
+the contributor's GitHub username, as well as the topic branch. 
+
+```bash
+cd jbosstools-server
+git remote add exampleuser git@github.com:exampleuser/jbosstools-server.git
+git fetch exampleuser
+git checkout prtopicbranch
+```  
+
+At this point, the maintainer should either run the modified source in the Eclipse Runtime Workbench (link TODO) or 
+run the build via command line (link TODO) to verify the unit tests all pass. 
+
+Once the patch is verified as correct and acceptable, the committer should then merge the pull request in 
+a neat and orderly fashion, which is the topic we will cover next. 
+
+=== How to merge a pull request correctly
+
+*Do NOT click the big green button!*
+
+image::images/forbidden_merge.png[Do Not Auto Merge]
+
+In order to keep an orderly git commit log history, JBoss Tools generally prefers clean
+merges to using the automatic merge ability provided by GitHub. What this means in principle
+is that we generally prefer to `squash` our commits, `rebase` against the most recent commits, 
+and end up with one commit per issue. 
+
+If, for example, you or a contributor was working in a topic branch to solve JBIDE-99999, 
+and 4 commits were made in the process, the following steps would need to be performed. 
+For this example, let's assume that the given pull request is intended to be pushed to master. 
+
+```
+git fetch contributorid
+git checkout topicbranch
+git rebase master     # This ensures all interim commits are placed before us
+
+# This allows us to squash, re-order, or otherwise modify our commits during the rebase, squash the last 3 commits into the first
+git rebase -i HEAD~5  
```

Comment:
I don't see what to change here. The paragraph directly before the commands says 

"If, for example, you or a contributor was working in a topic branch to solve JBIDE-99999, and 4 commits were made in the process, the following steps would need to be performed. For this example, let's assume that the given pull request is intended to be pushed to master. "

It basically already has as a precondition that you're working in a topic branch that is eventually intended for master.